tests/test_updater_validation.py

```
@@ -0,0 +1,103 @@
+        os.mkdir(self.metadata_dir)
+        os.mkdir(self.targets_dir)
+
+        # Setup the repository, bootstrap client root.json
+        self.sim = RepositorySimulator()
+        with open(os.path.join(self.metadata_dir, "root.json"), "bw") as f:
+            f.write(self.sim.signed_roots[0])
+
+    def tearDown(self) -> None:
+        self.temp_dir.cleanup()
+
+    def _new_updater(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            fetcher=self.sim,
+        )
+
+    def test_empty_target_path_validation(self) -> None:
+        updater = self._new_updater()
+        target_file = updater.get_targetinfo("")
+        self.assertIsNone(target_file)
+
+    def test_empty_targetinfo_validation(self) -> None:
+        # Initialize Updater without targets setup
+        updater = Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+        info = TargetFile(1, {"sha256": ""}, "")
+        with self.assertRaisesRegex(
+            ValueError, "target_dir must be set if filepath is not given"
+        ):
+            updater.download_target(info)
+
+        updater = self._new_updater()
+        with self.assertRaisesRegex(FetcherHTTPError, "No target"):
+            updater.download_target(info)
+
+    def test_empty_filepath_validation(self) -> None:
+        self.sim.add_target("targets", b"content", "targetpath")
+        self.sim.targets.version += 1
+        self.sim.update_snapshot()
+
+        updater = self._new_updater()
+        target_info = updater.get_targetinfo("targetpath")
+        assert target_info is not None
+        with self.assertRaises(securesystemslib.exceptions.StorageError):
+            updater.download_target(target_info, filepath="")
+
+    def test_non_existing_metadata_dir_validation(self) -> None:
```

nitpick: I'd just call it `test_non_existing_metadata_dir` (because we don't validate anything: we just let it fail)